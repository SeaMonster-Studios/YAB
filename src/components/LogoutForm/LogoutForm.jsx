import React from 'react'
import axios from 'axios'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import PropTypes from 'prop-types'
import Cookies from 'js-cookie'
//
import * as authActionCreators from '../../store/actions/auth'

class RegisterUserForm extends React.Component {
  static propTypes = {
    logoutAuthenticatedUser: PropTypes.func.isRequired,
  }
  constructor(props) {
    super(props)

    this.handleLogout = this.handleLogout.bind(this)
  }
  async handleLogout(e) {
    e.preventDefault()
    try {
      axios.get('/auth/logout')
      Cookies.remove('access_token')
      this.props.logoutAuthenticatedUser()
    } catch (error) {
      // SENTRY handle this
      console.log(error)
    }
  }
  render() {
    return (
      <button onClick={this.handleLogout} data-testid="logout-form-button">
        Logout
      </button>
    )
  }
}

const mapStateToProps = () => ({})

const mapDispatchToProps = dispatch =>
  bindActionCreators(authActionCreators, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(RegisterUserForm)
