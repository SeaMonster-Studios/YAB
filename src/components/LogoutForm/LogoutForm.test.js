import React from 'react'
import faker from 'faker'
//
import { renderWithRedux } from '../../utils/testing/testing'
import LogoutForm from './LogoutForm'

describe('LogoutForm Component Test', () => {
  it('Renders with correct default values from redux store', () => {
    const wrapper = renderSetupWithRedux()

    // make sure it renders
    expect(wrapper.container).toMatchSnapshot()

    // User should be logged in if this form is showing
    expect(wrapper.initialState.auth.authed).toBe(true)
    expect(wrapper.initialState.auth.user.email).toBeTruthy()
    expect(wrapper.initialState.auth.user.id).toBeTruthy()
  })
})

function renderSetupWithRedux(overrides) {
  const wrapper = renderWithRedux(<LogoutForm />)
  const logoutButton = wrapper.getByTestId('logout-form-button')
  const initialState = {
    ...wrapper.store.getState(),
    auth: {
      authed: true,
      user: {
        email: faker.internet.email(),
        id: faker.random.uuid(),
      },
    },
  }

  return {
    ...wrapper,
    logoutButton,
    initialState,
    ...overrides,
  }
}
