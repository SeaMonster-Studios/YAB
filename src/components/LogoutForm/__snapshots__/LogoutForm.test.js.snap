// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutForm Component Test Renders with correct default values from redux store 1`] = `
<div>
  <button
    data-testid="logout-form-button"
  >
    Logout
  </button>
</div>
`;
